### DISCUSSION TITLE
Unable to add network via 'WiFi Client' UI on wlan2

### DISCUSSION THREAD

**Discussion by done7k** [RaspAP/raspap-webgui#1229] [Q&A]
I have a fresh installation on RPI Zero W with 2nd Wifi dongle connected via USB as wlan1. the interface is seen on the 'WiFi Client' page in 'wlan1 up' states. I also can see available networks. When I try to connect to any of available hotspot (open of passkey) - enter passphrase and press 'Add' it returns 'WPA passphrase must be between 8 and 63 characters' message. I cannot see any change in /etc/wpa_supplicant/wpa_supplicant.conf file. And I suspect the error I receive constantly does not relate to a passphrase (at any case I use more than 8 chars) . Then what is the problem?

**Comment by billz** [comment]
If you're receiving a passphrase message, one of two things are likely: 1) the network you are connecting to is not "open", ie. it requires authentication, or 2) the passphrase entered does not comply with the IEEE standard used by RaspAP https://docs.raspap.com/faq/#passphrase

See also https://github.com/RaspAP/raspap-webgui/pull/1050 & related discussion.

**Reply by done7k** [reply]
Maybe I composed my question a wrong way, sorry. I will try to re-phrase it.
I want to have wlan0 in AP mode and set wlan1 being a gateway to the net. I attempt to connect wlan1 to an external WIFI spot. there are a least two types external wifi AP available: a) open with WEB authorizations and b) regular with passphrase. 
- If I use RaspAP UI to connect to any of such wifi's I got the 'WPA passphrase must be between 8 and 63 characters' message.
- Then I try to add the passphrase protected wifi into the /etc/wpa_supplicant/wpa_supplicant.conf file. And connect to it from a command line, I got it connected:
`wpa_supplicant -iwlan1 -c/etc/wpa_supplicant/wpa_supplicant.conf -d`
- I need to set metric for wlan0 and wlan1 correctly into the file 
`sudo nano /etc/dhcpcd.conf`
and it works fine. But I could not set all these via UI.

And there is an open question how do I set all this with a WIFI network which is open but require WEB authentication?

**Reply by billz** [reply]
You refer twice to "WEB" authentication. Do you mean WEP?

**Reply by done7k** [reply]
I am not quite sure the webauth is same as WEP. it is a guest hotspot in public places like Starbucks or airports. Please, take a look at the [Cisco doc](https://www.cisco.com/c/en/us/td/docs/wireless/controller/9800/config-guide/b_wl_16_10_cg/wireless-web-authentication.html)

**Reply by billz** [reply]
You're describing captive portals on open networks, which is discussed in #1039. This is a common use case for RaspAP.

I can only guess why you're unable to connect to those networks. Possibly they're returning an unexpected response that RaspAP doesn't know how to handle. If you can provide some useful debugging info I might be able to look into it, but no promises.

**Reply by done7k** [reply]
How can I turn on logging for the wpa_supplicant ? In syslog I could not find anything useful..

**Reply by done7k** [reply]
Made small change to the configure_client.php file to check 
`strlen($network['passphrase']) <= 64`
 instead of 63. And it added a network to wpa_supplicant.conf. Then I reverted the change in the php and it works for any other network... and 'open' one as well.. wierd
